### ISSUE TITLE
support for newpage

### ISSUE THREAD

**Issue by alphatan** [skuro/plantuml-mode#69]
# Is it possible to support the '**newpage**' command of plantuml in preview buffer ?

> In version 1.2.3, it's only showing the first page in the \*PLANTUML Preview\* buffer. 
> Is it possible to support the '**newpage**' command to show multiple page in the preview buffer?
> example code:
``` plantuml
@startuml


title page1

newpage 
title page2



@enduml
```
> The result only shows 'page1' 
>  
> Maybe it could consider following options:
> 1. Generate all pages and concatenate all into one buffer
> 2. Generate different pages in diffrent buffers.
>
>
> Thanks,

# Checklist for pull requests

- [x] I'm fine with my code being released under the GPLv2 license
- [x] the pull request is filed against the `develop` branch instead of `master`

**Comment by alphatan** [comment]
Duplicated for #59

**Comment by lockywolf** [comment]
Which version is this patch go to?

**Comment by skuro** [comment]
@lockywolf there is no patch available yet for #59

**Comment by amb007** [comment]
Here is an override that works for me (search for `AMB changed`).

```
;; override to support newpage
(defun org-babel-execute:plantuml (body params)
  "Execute a block of plantuml code with org-babel.
This function is called by `org-babel-execute-src-block'."
  (let* ((do-export (member "file" (cdr (assq :result-params params))))
         (out-file (if do-export
                       (or (cdr (assq :file params))
                           (error "No :file provided but :results set to file. For plain text output, set :results to verbatim"))
		     (org-babel-temp-file "plantuml-" ".txt")))
	 (cmdline (cdr (assq :cmdline params)))
         (in-file (make-temp-file       ;AMB changed
                   (concat (file-name-as-directory (org-babel-temp-directory))
                           (or (cdr (assq :file params)) "plantuml"))
                   nil
                   nil))
	 (java (or (cdr (assq :java params)) ""))
	 (executable (cond ((eq org-plantuml-exec-mode 'plantuml) org-plantuml-executable-path)
			   (t "java")))
	 (executable-args (cond ((eq org-plantuml-exec-mode 'plantuml) org-plantuml-args)
				((string= "" org-plantuml-jar-path)
				 (error "`org-plantuml-jar-path' is not set"))
				((not (file-exists-p org-plantuml-jar-path))
				 (error "Could not find plantuml.jar at %s" org-plantuml-jar-path))
				(t `(,java
				     "-jar"
				     ,(shell-quote-argument (expand-file-name org-plantuml-jar-path))
                                     ,@org-plantuml-args))))
	 (full-body (org-babel-plantuml-make-body body params))
	 (cmd (mapconcat #'identity
			 (append
			  (list executable)
			  executable-args
			  (pcase (file-name-extension out-file)
			    ("png" '("-tpng"))
			    ("svg" '("-tsvg"))
			    ("eps" '("-teps"))
			    ("pdf" '("-tpdf"))
			    ("tex" '("-tlatex"))
			    ("vdx" '("-tvdx"))
			    ("xmi" '("-txmi"))
			    ("scxml" '("-tscxml"))
			    ("html" '("-thtml"))
			    ("txt" '("-ttxt"))
			    ("utxt" '("-utxt")))
			  (list
			   cmdline      ;AMB changed
                           (org-babel-process-file-name in-file)
                           "-o"
                           (file-name-directory (buffer-file-name))))
			 " ")))
    (with-temp-file in-file (insert full-body))
    (message "%s" cmd) (org-babel-eval cmd "")
    (if (and (string= (file-name-extension out-file) "svg")
             org-babel-plantuml-svg-text-to-path)
        (org-babel-eval (format "inkscape %s -T -l %s" out-file out-file) ""))
    (unless do-export (with-temp-buffer
                        (insert-file-contents out-file)
                        (buffer-substring-no-properties
                         (point-min) (point-max))))))
```
